adapters/ix/ix.go

```diff
@@ -1,267 +1,189 @@
+		request.Site = &site
 	}
 
-	if len(bids) == 0 {
-		return nil, err
-	}
-	return bids, nil
-}
-
-func (a *IxAdapter) callOne(ctx context.Context, reqJSON bytes.Buffer) (ixBidResult, error) {
-	var result ixBidResult
+	// 'adapters/bidder.go': Bidder implementations may safely assume that this
+	// JSON has been validated by their static/bidder-params/{bidder}.json file.
+	var bidderExt adapters.ExtImpBidder
+	var ixExt openrtb_ext.ExtImpIx
+	json.Unmarshal(request.Imp[0].Ext, &bidderExt)
+	json.Unmarshal(bidderExt.Bidder, &ixExt)
 
-	httpReq, _ := http.NewRequest("POST", a.URI, &reqJSON)
-	httpReq.Header.Add("Content-Type", "application/json;charset=utf-8")
-	httpReq.Header.Add("Accept", "application/json")
+	request.Site.Publisher.ID = ixExt.SiteId
+}
 
-	ixResp, err := ctxhttp.Do(ctx, a.http.Client, httpReq)
-	if err != nil {
-		return result, err
+func getBannerFormats(banner *openrtb.Banner) []openrtb.Format {
+	if len(banner.Format) == 0 && banner.W != nil && banner.H != nil {
+		banner.Format = []openrtb.Format{{W: *banner.W, H: *banner.H}}
 	}
+	return banner.Format
+}
 
-	result.StatusCode = ixResp.StatusCode
-
-	if ixResp.StatusCode == http.StatusNoContent {
-		return result, nil
-	}
+func createRequestData(a *IxAdapter, request *openrtb.BidRequest, headers *http.Header) (*adapters.RequestData, error) {
+	body, err := json.Marshal(request)
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     a.URI,
+		Body:    body,
+		Headers: *headers,
+	}, err
+}
 
-	if ixResp.StatusCode == http.StatusBadRequest {
-		return result, &errortypes.BadInput{
-			Message: fmt.Sprintf("HTTP status: %d", ixResp.StatusCode),
+func (a *IxAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if glog.V(3) {
+		glog.Infof("bid-response: status-code: %d, body: %s", response.StatusCode, response.Body)
+	}
+	switch {
+	case response.StatusCode == http.StatusNoContent:
+		return nil, nil
+	case response.StatusCode == http.StatusBadRequest:
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	case response.StatusCode != http.StatusOK:
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	}
+
+	var bidResponse openrtb.BidResponse
+	if err := json.Unmarshal(response.Body, &bidResponse); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("JSON parsing error: %v", err),
+		}}
+	}
+
+	// Until the time we support multi-format ad units, we'll use a bid request impression media type
+	// as a bid response bid type. They are linked by the impression id.
+	impMediaType := map[string]openrtb_ext.BidType{}
+	for _, imp := range internalRequest.Imp {
+		if imp.Video != nil {
```

Comment:
How does your server handle a request for both video and banner?

Reply:
Our server will respond with banner. Fixed.